src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -318,4 +318,13 @@ object messages {
         |$code2
         |""".stripMargin
   }
+
+  case class IllegalExpressionStart()(implicit ctx: Context) extends Message(14) {
```

Comment:
I would call this something like: `IllegalStartSimpleExpr` since it is simple expressions and not all valid expressions. I'm assuming you chose 14 to coordinate with Shane's PR? :)